### PR TITLE
Add RBAC to Assist endpoints

### PR DESCRIPTION
**Pull request by jakule**
This PR adds RBAC to all assistant-related APIs

Closes https://github.com/gravitational/teleport/issues/25084

### CODE REVIEW

File: lib/auth/auth_with_roles.go [hunk 1]
```diff
@@ -5957,31 +5957,46 @@ func (a *ServerWithRoles) UpdateHeadlessAuthenticationState(ctx context.Context,
...
+		return nil, trace.Wrap(err)
+	}
+
 	return a.authServer.GetAssistantMessages(ctx, id)
 }
 
 // InsertAssistantMessage adds the message to the backend.
 func (a *ServerWithRoles) InsertAssistantMessage(ctx context.Context, msg *proto.AssistantMessage) (*emptypb.Empty, error) {
-	// TODO(jakule): Check if user has access to given conversation ID
+	if err := a.action(apidefaults.Namespace, types.KindAssistant, types.VerbCreate); err != nil {
```
Comment (r0mant) [inline lib/auth/auth_with_roles.go:5987]: Nit: Wouldn't `VerbUpdate` be a little more appropriate here? You're updating assistant conversation by adding a new message?

File: lib/auth/auth_with_roles.go [hunk 2]
```diff
@@ -5957,31 +5957,46 @@ func (a *ServerWithRoles) UpdateHeadlessAuthenticationState(ctx context.Context,
...
+		return nil, trace.Wrap(err)
+	}
+
 	return a.authServer.CreateAssistantConversation(ctx, req)
 }
 
 // GetAssistantConversations returns all conversations started by a user.
 func (a *ServerWithRoles) GetAssistantConversations(ctx context.Context, request *proto.GetAssistantConversationsRequest) (*proto.GetAssistantConversationsResponse, error) {
-	// TODO(jakule): Check if user has access to given conversation ID
+	if err := a.action(apidefaults.Namespace, types.KindAssistant, types.VerbList); err != nil {
```
Comment (r0mant) [inline lib/auth/auth_with_roles.go:5969]: In these get/update methods, should we allows users to read/update their own conversations? Like we use `currentUserAction` in some other RBAC checks to allow users update their own passwords or read their own access requests for example.

Comment (jakule) [reply lib/auth/auth_with_roles.go:5969]: This is a bit different case. Currently, users have only access to their own conversations.
We want to keep this feature enabled by default, but as an administrator, I think you should have the option to disable access for some users. In this case, we should disable access to all APIs, and users with disabled access should not able able to even list their own conversations.

File: lib/auth/auth_with_roles.go (hunk 1 above)
Comment (jakule) [reply lib/auth/auth_with_roles.go]: I was thinking more about it that a message is being added, hence `create`, but I agree that `updating` conversation is probably more intuitive.
Changed.

**Review by r0mant — APPROVED** [review]

**Review by xacrimon — APPROVED** [review]

**Pull request merged**